### DISCUSSION TITLE
Whitelist GeoSPARQL 1.1 Aggregate Operators

### DISCUSSION THREAD

**Discussion by apano-on** [eclipse-rdf4j/rdf4j#4986] [Q&A]
GeoSPARQL 1.1 was [recently released](https://www.ogc.org/standard/geosparql/) and it introduced [6 new](https://docs.ogc.org/is/22-047r1/22-047r1.html#sa_functions) geospatial aggregation functions such as geof:aggUnion. However, when parsing a geospatial query involving aggregation with a ```GROUP BY``` clause using the ```rdf4j-queryparser-sparql``` module, RDF4J will throw an error such as e.g.:
 ```
non-aggregate expression 'FunctionCall (http://www.opengis.net/def/function/geosparql/aggunion)
   Var (name=xWKT)
' not allowed in projection when using GROUP BY.
```

Is it possible to whitelist these 6 functions and add them to the collection of allowed aggregate operators in the official release together with AVG/SUM/COUNT? If I am not mistaken the AGG prefix and certain [aggregation functions](https://github.com/eclipse-rdf4j/rdf4j/blob/main/core/queryparser/sparql/src/main/java/org/eclipse/rdf4j/query/parser/sparql/TupleExprBuilder.java#L2563) seem to be already whitelisted. I would probably consider GeoSPARQL to deserve the same degree of official support.

**Comment by frensjan** [comment]
As I understand it there would be more to it than just whitelisting the function. `aggUnion` nor any of it's 'siblings' are mentioned in https://rdf4j.org/documentation/programming/geosparql/. Also, there isn't an implementation for it in https://github.com/eclipse-rdf4j/rdf4j/tree/main/core/queryalgebra/geosparql/.

An implementation of `org.eclipse.rdf4j.query.parser.sparql.aggregate.AggregateFunction` and `org.eclipse.rdf4j.query.parser.sparql.aggregate.AggregateFunctionFactory` would be required for this to work.
